extensions/mssql/src/objectManagement/ui/loginDialog.ts

```diff
@@ -132,6 +132,11 @@ export class LoginDialog extends PrincipalDialogBase<ObjectManagement.Login, Obj
 		const items: azdata.Component[] = [];
 		this.passwordInput = this.createPasswordInputBox(objectManagementLoc.PasswordText, async (newValue) => {
 			this.objectInfo.password = newValue;
+			this.mustChangePasswordCheckbox.enabled = this.objectInfo.enforcePasswordPolicy && this.isPasswordChanged();
+			// this handles the case where the mustChangePasswordCheckbox is disabled when a user changes the password input and reverts the change. In that case we want to reset the check state of this checkbox to its original value instead of using the potentially dirty state set during password input changes.
```

Comment:
This behavior is different from SSMS behavior. When SSMS disables the checkbox due to password field change, it doesn't reset the check state. If I modify the check state between password changes, that dirty state is kept.